sdk/core/core/src/az_span_builder.c

```diff
@@ -61,3 +61,15 @@ AZ_NODISCARD az_result az_span_builder_replace(
   self->length = size_after_replace;
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result
+az_span_builder_append_zeros(az_span_builder * const self, size_t const size) {
+  AZ_CONTRACT_ARG_NOT_NULL(self);
+
+  az_mut_span const span = az_mut_span_take(az_mut_span_drop(self->buffer, self->length), size);
+  if (span.size != size) {
+    return AZ_ERROR_BUFFER_OVERFLOW;
```

Comment:
validity of builder after this failure?

Reply:
the buffer would be fine. We don't change it.